src/sql/base/browser/ui/table/plugins/autoSizeColumns.plugin.ts

```diff
@@ -134,13 +147,40 @@ export class AutoColumnSize<T extends Slick.SlickData> implements Slick.Plugin<T
 		}
 	}
 
+	private getMaxColumnTextWidths(columnDefs: Slick.Column<T>[], colIndices: number[]): number[] {
+		let data = this._grid.getData() as Slick.DataProvider<T>;
+		let viewPort = this._grid.getViewport();
+		let start = Math.max(0, viewPort.top);
+		let end = Math.min(data.getLength(), 10); // Limit the number of rows we check to 10
```

Comment:
In query editor, the result grid viewport is usually around 10 rows but in notebooks, it can go up to 20 rows. In most cases I think 10 rows should be enough.